odoo/models.py

```diff
@@ -4506,6 +4518,61 @@ def _parent_store_update(self):
         self.env.cache.update(records, self._fields['parent_path'], updated.values())
         records.modified(['parent_path'])
 
+    def _identify_existing_records(self, data_list, search_fields):
+        ''' Find or create an External ID (xml_id) for each of the unique records
+            having matching data in search_fields
+
+            :param data_list: list of dicts with keys `xml_id` and record values
+            :param search_fields: list of technical field names
+
+            :return: updated data_list containing the new/found xml_id
+
+            :raises ValidationError: when multiple records match the search criteria
+        '''
```

Comment:
- needs to be rewritten: it's not accurate anymore
- would be nice to say that those ids are added in the `data_list` too